impeller/toolkit/android/surface_control.h

```diff
@@ -0,0 +1,86 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_TOOLKIT_ANDROID_SURFACE_CONTROL_H_
+#define FLUTTER_IMPELLER_TOOLKIT_ANDROID_SURFACE_CONTROL_H_
+
+#include "flutter/fml/unique_object.h"
+#include "impeller/toolkit/android/proc_table.h"
+
+namespace impeller::android {
+
+//------------------------------------------------------------------------------
+/// @brief      A wrapper for ASurfaceControl.
+///             https://developer.android.com/ndk/reference/group/native-activity#asurfacecontrol
+///
+///             Instances of this class represent a node in the hierarchy of
+///             surfaces sent to the system compositor for final composition.
+///
+///             This wrapper is only available on Android API 29 and above.
+///
+class SurfaceControl {
```

Comment:
Does the surface control have any threading restrictions? In general, if we're aware of any threading restrictions for parts of these APIs it would be good to note them.

Reply:
I am not aware of any threading restrictions on any of the handle types. The only major quirk is the transaction completion callback being made on a system managed thread (presumably a binder thread). I've documented that and also the lifecycle implications to the transaction because of it in the headerdoc comment for it.

Reply:
I audited all calls, and beyond the completion callback that I have already documented, there is little guidance from the Android docs about the threading restrictions of any of the calls. I'm just going to be safe and call these as needing to be externally synchronized.